File: tests/fixtures/parity/shapes.cs
class Shapes {
    int flatCalls() {
        int a = 1;
        int b = 2;
        int c = a + b;
        return c;
    }

    int pickBranch(int x) {
        if (x > 10) {
            return 3;
        } else if (x > 5) {
            return 2;
        } else {
            return 1;
        }
    }

    int nestedGuard(int a, int b, int c) {
        if (a > 0) {
            if (b > 0) {
                if (c > 0) {
                    return 1;
                }
            }
        }
        return 0;
    }

    int loopFilter(int n) {
        int total = 0;
        for (int i = 0; i < n; i++) {
            if (i % 2 == 0) {
                total += i;
            }
        }
        return total;
    }

    int wideParams(int a, int b, int c, int d, int e, int f) {
        return a + b + c + d + e + f;
    }

    int boolBlend(int a, int b, int c, int d) {
        if ((a > 0 && b > 0) || (c > 0 && d > 0)) {
            return 1;
        }
        return 0;
    }
}

File: tests/fixtures/parity/Shapes.hs
module Shapes where

flatCalls :: Int -> Int
flatCalls x =
  let a = 1
      b = 2
      c = a + b
  in c + x

pickBranch :: Int -> Int
pickBranch x =
  if x > 10
    then 3
    else if x > 5
      then 2
      else 1

nestedGuard :: Int -> Int -> Int -> Int
nestedGuard a b c =
  if a > 0
    then if b > 0
      then if c > 0
        then 1
        else 0
      else 0
    else 0

loopFilter :: [Int] -> Int
loopFilter items = go items 0
  where
    go [] total = total
    go (x:xs) total =
      if x > 0
        then go xs (total + x)
        else go xs total

wideParams :: Int -> Int -> Int -> Int -> Int -> Int -> Int
wideParams a b c d e f = a + b + c + d + e + f

boolBlend :: Int -> Int -> Int -> Int -> Int
boolBlend a b c d =
  if (a > 0 && b > 0) || (c > 0 && d > 0)
    then 1
    else 0
